File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        int checkAdd=0;
        for(int i=1;i<3;i++){
            if(i==1)
                checkAdd=common[i]-common[i-1];
            else{
                if(checkAdd==common[i]-common[i-1])
                    answer=common[common.length-1]+checkAdd;
                else{
                    for(int j=0;j<common.length;j++){
                        if(common[i-1]!=0 && common[i]!=0){
                            answer=common[common.length-1]*(common[i]/common[i-1]);
                            break;
                        }
                    }
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.py
def solution(common):
    answer = 0
    for i in range(1,3):
        if i==1:
            checkAdd=common[i]-common[i-1]
        else:
            if checkAdd==common[i]-common[i-1]:
                answer=common[-1]+checkAdd
            else:
                for i in range(1,len(common)):
                    if common[i-1]!=0 and common[i]!=0:
                        answer=common[-1]*(common[i]//common[i-1])
                        break
    return answer